File: src/TinyHttpResponse.scala

class TinyHttpResponse(statusCode: Int) {

  var content: String = ""
  var headers: Map[String, String] = Map[String, String]()

  def addHeader(key: String, value: String): Unit = {
    headers += key -> value
  }


  def responseMessage: String = {
    val statusText: String = Constants.HTTP_RESPONSE_CODE_MESSAGE_LOOKUP.getOrElse(statusCode, "Unknown")

    val headerLines = headers.map { case (k, v) => k + ": " + v }
    var headerText = ""
    if (headerLines.nonEmpty) {
      headerText = headerLines reduce((line1, line2) => line1 + "\n" + line2)
    }

    val contentLength:Int = content.length

    s"HTTP/1.1 $statusCode $statusText\n$headerText\nContent Length: $contentLength\n\n$content"
  }
}

File: src/tiny-http-response.js
import _ from 'lodash';

export class TinyHttpResponse {
    constructor() {
        this.headers = {};
    }

    status(code) {
        this.statusCode = code;
    }

    body(data) {
        this.body = data;
    }

    addHeader(key, value) {
        this.headers[key] = value;
    }

    getResponseMessage() {
        const statusText = 'OK';

        const headerLines = _.mapKeys(this.headers, (val, key) => {
            return `${key}:${val}`;
        });

        return `HTTP/1.1 ${this.statusCode} ${statusText}\n\n ${this.body}`;
    }
}
